dawn/src/dawn/IIR/DependencyGraphAccesses.cpp

```diff
@@ -618,13 +616,17 @@ bool DependencyGraphAccesses::exceedsMaxBoundaryPoints(int maxHorizontalBoundary
   std::unordered_map<std::size_t, Extents> extentMap = computeBoundaryExtents(this);
 
   for(const auto& vertexIDExtentsPair : extentMap) {
-    const auto& hExtents = dawn::iir::extent_cast<dawn::iir::CartesianExtent const&>(
-        vertexIDExtentsPair.second.horizontalExtent());
-    if(hExtents.iPlus() > maxHorizontalBoundaryExtent ||
-       hExtents.iMinus() < -maxHorizontalBoundaryExtent ||
-       hExtents.jPlus() > maxHorizontalBoundaryExtent ||
-       hExtents.jMinus() < -maxHorizontalBoundaryExtent)
+    if(extent_dispatch(vertexIDExtentsPair.second.horizontalExtent(),
+                       [&](iir::CartesianExtent const& hExtent) {
+                         return hExtent.iPlus() > maxHorizontalBoundaryExtent ||
+                                hExtent.iMinus() < -maxHorizontalBoundaryExtent ||
+                                hExtent.jPlus() > maxHorizontalBoundaryExtent ||
+                                hExtent.jMinus() < -maxHorizontalBoundaryExtent;
+                       },
+                       [](iir::UnstructuredExtent const&) { return false; },
```

Comment:
I created an issue for this to have the function argument as an extent. It might be possible that we don't even run this code for unstructured right now.  If I really need to, I will put a hack instead of `extend_cast`.